packages/ckeditor5-engine/tests/conversion/upcastdispatcher.js

```diff
@@ -605,6 +605,39 @@ describe( 'UpcastDispatcher', () => {
 				sinon.assert.calledOnce( spy );
 			} );
 
+			it( 'should insert paragraph if it\'s allowed and if it would allow conversion', () => {
+				const spy = sinon.spy();
+
+				model.schema.register( 'section', {
+					allowIn: '$root'
+				} );
+				model.schema.register( 'span', {
+					allowIn: 'paragraph'
+				} );
+				model.schema.extend( 'paragraph', {
+					allowIn: 'section'
+				} );
+
+				dispatcher.on( 'documentFragment', ( evt, data, conversionApi ) => {
+					const section = conversionApi.writer.createElement( 'section' );
+					const position = ModelPosition._createAt( section, 0 );
```

Comment:
could you use `conversionApi.writer` to create postions? Using `writer` or `model` is preferable then importing `ModelPosition`.
We left that in tests and in engine source code but we might use proper API in the test.